File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        //System.out.println(s.length());
        int[] arr = new int[128];
        int ans = 0, start = 0, count = 0;
        for(int end=0;end<s.length();end++){
            while(arr[s.charAt(end)]==1){
                arr[s.charAt(start)] = 0;
                start++;
                count--;
            }
            arr[s.charAt(end)] = 1;
            count++;
            ans = Math.max(ans,count);
        }
        return ans;
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        n = len(s)
        if n==0:
            return n
        dict1 = {s[0]:1}
        ans,count = 1,1
        i = 0
        j = 1
        while j<n:
            if s[j] not in dict1:
                dict1[s[j]] = 1
                count += 1
                j += 1
            else:
                ans = max(count,ans)
                count -= 1
                dict1.pop(s[i])
                i += 1
        ans = max(ans,count)
        return ans
        
        